vmicore/src/os/windows/KernelAccess.cpp

```diff
@@ -194,23 +237,71 @@ addr_t KernelAccess::getMmSectionFlagsAddr(addr_t controlAreaBaseVA) const
 
 bool KernelAccess::extractIsBeingDeleted(addr_t controlAreaBaseVA) const
 {
-    return static_cast<bool>(extractFlagValue32(getMmSectionFlagsAddr(controlAreaBaseVA),
-                                                kernelOffsets.mmsectionFlags.beingDeleted.startBit,
-                                                kernelOffsets.mmsectionFlags.beingDeleted.endBit));
+    auto flagStructSize = vmiInterface->getStructSizeFromJson(KernelStructOffsets::mmsection_flags::structName);
+    assert((kernelOffsets.mmsectionFlags.beingDeleted.endBit - kernelOffsets.mmsectionFlags.beingDeleted.startBit) ==
+           1);
+    switch (flagStructSize)
+    {
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+        case 4:
```

Comment:
why not:
```suggestion
        case sizeof(uint32_t):
```

Reply:
In my opinion '4' has better readability